DietPi-Software | Tailscale VPN

goldfix opened the pull request:
Added tailscale VPN

Tested on:
x86_64/amd64
armv6l/armhf
armv7l/armhf

goldfix committed:
added tailscale un/install procedures

goldfix committed:
added filter for specific distro

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -7971,6 +7980,17 @@ _EOF_
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then
+
+			Banner_Installing
+			G_EXEC curl -fsSL https://pkgs.tailscale.com/stable/debian/bullseye.noarmor.gpg -o /usr/share/keyrings/tailscale-archive-keyring.gpg
```
there you might need to detect the Debian version as we support Buster, Bullseye as well as Bookworm

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -7971,6 +7980,17 @@ _EOF_
+
+			Banner_Installing
+			G_EXEC curl -fsSL https://pkgs.tailscale.com/stable/debian/bullseye.noarmor.gpg -o /usr/share/keyrings/tailscale-archive-keyring.gpg
+			G_EXEC curl -fsSL https://pkgs.tailscale.com/stable/debian/bullseye.tailscale-keyring.list -o /etc/apt/sources.list.d/tailscale.list
```
Same on this one, Debian version would need to be detected.

goldfix committed:
added to detect Debian version

goldfix committed:
fix check distro variables

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -1396,8 +1396,12 @@ Available commands:
 		aSOFTWARE_CATX[$software_id]=15
 		aSOFTWARE_DOCS[$software_id]='https://dietpi.com/docs/software/vpn/#tailscale'
 		aSOFTWARE_INTERACTIVE[$software_id]=1
+		# Disable support for all distro
```
that's not needed at all as there is no other distro we support /being available. On Stretch we don't do any updates anymore. Means it can be enabled for distro.

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -7984,8 +7988,26 @@ _EOF_
+				G_EXEC curl -fsSL https://pkgs.tailscale.com/stable/debian/bookworm.noarmor.gpg -o /usr/share/keyrings/tailscale-archive-keyring.gpg
+				G_EXEC curl -fsSL https://pkgs.tailscale.com/stable/debian/bullseye.tailscale-keyring.list -o /etc/apt/sources.list.d/tailscale.list
+			fi
+
```
You could stick to the original version. Just replace the distro name with a variable like `$G_DISTRO_NAME`

goldfix committed:
removed distro checking

goldfix committed:
used G_DISTRO_NAME variable

goldfix committed:
Merge branch 'dev' into tailscale_v1

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -7971,6 +7978,28 @@ _EOF_
+
+			G_AGUP
+			G_AGI tailscale
+		fi
```
This is way more simplified 

```
		software_id=201 # Tailscale
		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then

			Banner_Installing

			# APT key
			G_EXEC eval "curl -sSLf https://pkgs.tailscale.com/stable/debian/$G_DISTRO_NAME.gpg | gpg --dearmor -o /etc/apt/trusted.gpg.d/dietpi-tailscale.gpg --yes"

			# APT list
			G_EXEC eval "echo 'deb https://pkgs.tailscale.com/stable/debian $G_DISTRO_NAME main' > /etc/apt/sources.list.d/tailscale.list"

			G_AGUP
			G_AGI tailscale
		fi
```

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -14240,6 +14269,16 @@ If no WireGuard (auto)start is included, but you require it, please do the follo
+			G_AGUP
+			G_AGP tailscale
+			G_EXEC rm -rf /var/lib/tailscale/tailscaled.state
+
```
key as well as source list to be removed

```
			[[ -f '/etc/apt/sources.list.d/tailscale.list' ]] && G_EXEC rm /etc/apt/sources.list.d/tailscale.list
			[[ -f '/etc/apt/trusted.gpg.d/dietpi-tailscale.gpg' ]] && G_EXEC rm /etc/apt/trusted.gpg.d/dietpi-tailscale.gpg
```

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -14240,6 +14269,16 @@ If no WireGuard (auto)start is included, but you require it, please do the follo
+			Banner_Uninstalling
+			G_AGUP
+			G_AGP tailscale
+			G_EXEC rm -rf /var/lib/tailscale/tailscaled.state
```
check the availibilty of the file before trying to remove

```
[[ -f '/var/lib/tailscale/tailscaled.state' ]] && G_EXEC rm -Rf /var/lib/tailscale/tailscaled.state
```

goldfix committed:
simplified install scripts

goldfix committed:
substituted and added deletion steps with conditional version

goldfix committed:
added service name into 'dietpi-services'

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -14240,6 +14262,18 @@ If no WireGuard (auto)start is included, but you require it, please do the follo
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == -1 )); then
+
+			Banner_Uninstalling
+			G_AGUP
```
It's not necessary to update package lists before purging a package:
```suggestion
```

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -14240,6 +14262,18 @@ If no WireGuard (auto)start is included, but you require it, please do the follo
+			Banner_Uninstalling
+			G_AGUP
+			G_AGP tailscale
+			[[ -f '/var/lib/tailscale/tailscaled.state' ]] && G_EXEC rm /var/lib/tailscale/tailscaled.state
```
```suggestion
			[[ -d '/var/lib/tailscale' ]] && G_EXEC rm -R /var/lib/tailscale
```

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -7971,6 +7978,21 @@ _EOF_
+			G_EXEC eval "curl -sSLf https://pkgs.tailscale.com/stable/debian/$G_DISTRO_NAME.gpg | gpg --dearmor -o /etc/apt/trusted.gpg.d/dietpi-tailscale.gpg --yes"
+
+			# APT list
+			G_EXEC eval "echo 'deb https://pkgs.tailscale.com/stable/debian $G_DISTRO_NAME main' > /etc/apt/sources.list.d/tailscale.list"
```
Let's have files installed by `dietpi-software` clearly identifiable:
```suggestion
			G_EXEC eval "echo 'deb https://pkgs.tailscale.com/stable/debian $G_DISTRO_NAME main' > /etc/apt/sources.list.d/dietpi-tailscale.list"
```

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -14240,6 +14262,18 @@ If no WireGuard (auto)start is included, but you require it, please do the follo
+			G_AGUP
+			G_AGP tailscale
+			[[ -f '/var/lib/tailscale/tailscaled.state' ]] && G_EXEC rm /var/lib/tailscale/tailscaled.state
+			[[ -f '/etc/apt/sources.list.d/tailscale.list' ]] && G_EXEC rm /etc/apt/sources.list.d/tailscale.list
```
```suggestion
			[[ -f '/etc/apt/sources.list.d/tailscale.list' ]] && G_EXEC rm /etc/apt/sources.list.d/dietpi-tailscale.list
```

goldfix committed:
Update dietpi/dietpi-software

Co-authored-by: MichaIng <micha@dietpi.com>

goldfix committed:
Update dietpi/dietpi-software

Co-authored-by: MichaIng <micha@dietpi.com>

goldfix committed:
Update dietpi/dietpi-software

Co-authored-by: MichaIng <micha@dietpi.com>

goldfix committed:
Update dietpi/dietpi-software

Co-authored-by: MichaIng <micha@dietpi.com>

MichaIng committed:
Merge branch 'dev' into tailscale_v1

MichaIng committed:
v8.6

- DietPi-Software | Tailscale: Use software ID 58 and download de-armored key

MichaIng committed:
v8.6

- DietPi-Software | Tailscale: Enable IP forwarding for LAN access by default, align method with OpenVPN and WiFi Hotspot (where forced RA acceptance was missing)

MichaIng committed:
v8.6

- DietPi-Software | Tailscale: It can be installed non-interactively

MichaIng committed:
v8.6

- DietPi-Software | Tailscale: Add changelog, readme and survey report entry

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -2557,6 +2563,13 @@ _EOF_
+			echo -e 'net.ipv4.ip_forward=1\nnet.ipv6.conf.default.accept_ra=2\nnet.ipv6.conf.all.accept_ra=2\nnet.ipv6.conf.default.forwarding=1\nnet.ipv6.conf.all.forwarding=1' > "/etc/sysctl.d/dietpi-$1.conf"
+			sysctl net.ipv4.ip_forward=1 net.ipv6.conf.default.accept_ra=2 net.ipv6.conf.all.accept_ra=2 net.ipv6.conf.default.forwarding=1 net.ipv6.conf.all.forwarding=1
+		}
+
```
Should we at least give a notification line on this step? Just to give some information an what happen?

```
[  OK  ] DietPi-Software | APT install for: tailscale
net.ipv4.ip_forward = 1
net.ipv6.conf.default.accept_ra = 2
net.ipv6.conf.all.accept_ra = 2
net.ipv6.conf.default.forwarding = 1
net.ipv6.conf.all.forwarding = 1
```

MichaIng replied:
Done

MichaIng committed:
v8.6

- DietPi-Software | Tailscale: Add notification when enabling IP forwarding
